### ISSUE TITLE
Looking for best practice advice on removing a smelly pattern

### ISSUE THREAD

**Issue by rromanchuk** [getsentry/sentry-ruby#1710]
I noticed an annoying habit that whenever using `Sentry.capture_exception` or `Sentry.capture_message` that i'm adding additional lines to be confident i get the exception/message to STDOUT or logger.  For example...

```ruby
rescue MyError::SpecialCase
  Sentry.capture_message("My Message")
  DebugLogger.error "My message"
rescue StandardError => e
  ap e
  Sentry.capture_exception(e)
end
```

This is partly an artifact of setting `config.dsn = nil` in development, but there are times in a production context I want this as well. My logging in production is 100% structured, and when i'm consuming logs in a discovery/exploratory manner, there are cases where I know i'm missing resolution because I can see async payloads being delivered to sentry when I wish i had the structured dump right there without context/tool switching. 

Is using a `before_send` block the right thing to use here? My only concern is that this might be NOOP if DSN isn't set as the documentation sort of implies with "to Sentry"

> Provide a lambda or proc. This will be called before sending an event to Sentry. Receives an event and hint as parameter

TLDR: I want to use a single Sentry SDK call to both send to sentry and device IO, in all environments. Sort of related to [this](https://github.com/getsentry/sentry-javascript/issues/1600) 

You can auto close this, just looking for feedback
